src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncContext.Unix.cs

```diff
@@ -1986,7 +1986,8 @@ public SocketError SendFileAsync(SafeFileHandle fileHandle, long offset, long co
         // be scheduled instead. It's not functionally incorrect to schedule the release of a synchronous operation, just it may
         // lead to thread pool starvation issues if the synchronous operations are blocking thread pool threads (typically not
         // advised) and more threads are not immediately available to run work items that would release those operations.
-        public unsafe Interop.Sys.SocketEvents HandleSyncEventsSpeculatively(Interop.Sys.SocketEvents events)
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
```

Comment:
This alone makes a measurable impact?

Reply:
> This alone makes a measurable impact?

Yes, it does.

Reply:
How much?  I'm not against it (in particular because there's only one call site), but if this actually matters, that suggests the overhead of a single non-virtual, non-generic method call is measurable... and we have a lot of those.  I'm wondering why this one in particular is meaningful.

Reply:
This is the only method on the epoll thread that gets called per event. And when there are no sync-over-async operations (like TE benchmarks) it doesn't call other methods.

The improvement may be 'boosted' when the epoll thread is on the edge of being a bottleneck (like with the speculative optimization).